src/smtencoding/FStar.SMTEncoding.EncodeTerm.fs

```diff
@@ -392,6 +392,24 @@ let is_BitVector_primitive head args =
     | _ -> false
 
 
+module CBytes = FStar.Compiler.Bytes
+
+(*
+ * Hashcons the string constant
+ * We were earlier using a counter, which is not precise in the presence of
+ *   checked files, cf. #1999
+ *)
+let encode_string_const (s:string) : term =
+  let rec aux (acc:BigInt.bigint) (i:int) (bs:CBytes.bytes) =
+    if i = CBytes.length bs then acc
+    else
+      let acc = BigInt.add_big_int
+        (BigInt.mult_big_int (256 |> BigInt.of_int_fs) acc)
+        (CBytes.get bs i |> BigInt.of_int_fs) in
+      aux acc (i+1) bs in
+  let id = aux BigInt.zero 0 (CBytes.string_as_unicode_bytes s) in
```

Comment:
Doesn't this `string_as_unicode_bytes` return an array of unicode scalar values instead of bytes?

Reply:
Yeah.. here's a collision :)
```fstar
let s1 = "䅁"
let s2 = "AA"

let _ = assert (s1 == s2)
```